testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6618,6 +6618,34 @@ private static void checkIf(SqlOperatorFixture f) {
     f.checkNull("pow(cast(null as integer), 2)");
   }
 
+  @Test void testIsInfFunc() {
+    final SqlOperatorFixture f0 = fixture().setFor(SqlLibraryOperators.IS_INF);
+    f0.checkFails("^is_inf(3)^",
+        "No match found for function signature IS_INF\\(<NUMERIC>\\)",
+        false);
+    final SqlOperatorFixture f = f0.withLibrary(SqlLibrary.BIG_QUERY);
+    f.checkBoolean("is_inf(3)", false);
+    f.checkBoolean("is_inf(1.2345)", false);
+    f.checkBoolean("is_inf(cast('NaN' as double))", false);
+    f.checkBoolean("is_inf(cast('Infinity' as double))", true);
+    f.checkBoolean("is_inf(cast('-Infinity' as double))", true);
```

Comment:
did we ever test `CAST('Infinity' as DOUBLE)`?

is this standard SQL behavior?

Reply:
also check on REAL values (implicit cast of 32-bit float to 64-bit float)

Reply:
Added tests for REAL and Infinity/NaN